lib/search_files.js

```diff
@@ -0,0 +1,252 @@
+const fs = require('fs');
+const path = require('path');
+const oshomedir = require('os-homedir');
+
+const isFunction = require('lodash.isfunction');
+const isPlainObject = require('lodash.isplainobject');
+const isArray = require('lodash.isarray');
+const isString = require('lodash.isstring');
+const doNothing = require('lodash.identity');
+const mapValues = require('lodash.mapvalues');
+
+const isAbsolutePath = require('is-absolute');
+const parsePath = require('parse-filepath');
+
+const isWindows = process.platform === 'win32';
+
+
+function firstUnNullElem(arr, fn /*, ...*/) {
+  var args = Array.prototype.slice.call(arguments, 2);
+
+  for (var i = 0, n = arr.length; i < n; i++) {
+    var ret = fn.apply(arr, [arr[i]].concat(args));
+    if (ret) {
+      return ret;
+    }
+  }
+  return null;
+}
+
+function eachUnNullElem(arr, fn /*, ...*/) {
+  var args = Array.prototype.slice.call(arguments, 2);
+
+  var ret = [];
+  for (var i = 0, n = arr.length; i < n; i++) {
+    var v = fn.apply(arr, [arr[i]].concat(args));
+    if (v) {
+      if (isArray(v)) {
+        Array.prototype.push.apply(ret, v);
+      } else {
+        ret.push(v);
+      }
+    }
+  }
+  return ret;
+}
+
+
+function normalizeHomedir(pth, homedir) {
```

Comment:
more @jonschlinkert code!  should probably just use https://github.com/jonschlinkert/expand-tilde instead of a custom module here